packages/gatsby/src/schema/graphql-engine/bundle-webpack.ts

```diff
@@ -35,6 +37,71 @@ function getApisToRemoveForQueryEngine(): Array<GatsbyNodeAPI> {
   return apisToRemove
 }
 
+const getInternalPackagesCacheDir = (): string =>
+  path.join(process.cwd(), `.cache/internal-packages`)
```

Comment:
I do wonder if there is something we could do to "sandbox" this to not have potential repo/site level `.npmrc` (or even user level `.npmrc` locally) potentially cause problems with installing optional packages. If there is `.npmrc` with `omit=optional` (or however the syntax it is) - that would use in this directory as well?

Alternatively, maybe instead of installing whole `lmdb` (and let that install optional packages with prebuilds) - maybe we try to install those prebuild package directly? Maybe then it would be less prone to inherit same problem that cause those not to be installed normally in the first place?

Reply:
> Alternatively, maybe instead of installing whole lmdb (and let that install optional packages with prebuilds) - maybe we try to install those prebuild package directly?

I guess what I wonder is, in the presence of `@lmdb/x` package will that always be favoured even if there's a `build` from source in the `lmdb` directory? (I can dig through the code to try to understand that but tbh I worry that we might miss something by doing so)

Reply:
> I guess what I wonder is, in the presence of `@lmdb/x` package will that always be favoured even if there's a `build` from source in the `lmdb` directory? (I can dig through the code to try to understand that but tbh I worry that we might miss something by doing so)

It looks like general order of loading is to first try to load from `build/Release` first before trying optional packages. However this relies on fact that there is `install` script that before trying to compile from source check if there is binary that can be loaded - and when prebuilts are installed - it finds those and that check prevent from even compiling from source ( https://unpkg.com/browse/node-gyp-build-optional-packages/ is doing all those things ).

However - our `lmdb-bundling-patch` replace runtime usage of `node-gyp-build(-optional-packages)` from lmdb and replace that with direct path to `.node` file, so we are not subject to regular behavior when bundling for lambdas as we get rid of it here: https://github.com/gatsbyjs/gatsby/blob/22c24122f321d60011aec7daec86a1ccf89e994d/packages/gatsby/src/schema/graphql-engine/lmdb-bundling-patch.ts#L74-L85